adwords_client/internal_api/builder.py

```diff
@@ -54,45 +56,152 @@ def cast_operation(self, operation):
     def filter_operation(self, operation):
         return {k: v for k, v in operation.items() if v is not None}
 
-    def _parse_operation(self, operation):
+    def _parse_operation(self, operation, is_get_operation=False, sync=None):
         if self.valid_operation(operation):
-            object_type = operation.pop('object_type')
+            object_type = operation.get('object_type')
             operation = self.filter_operation(self.cast_operation(operation))
             if object_type == 'keyword':
-                yield from self._parse_keyword(operation)
+                yield from self._parse_keyword(operation, is_get_operation)
             elif object_type == 'adgroup':
-                yield from self._parse_adgroup(operation)
+                yield from self._parse_adgroup(operation, is_get_operation)
             elif object_type == 'ad':
-                yield from self._parse_ad(operation)
-            elif object_type == 'campaign':
-                yield from self._parse_campaign(operation)
+                yield from self._parse_ad(operation, is_get_operation)
+            elif object_type == 'campaign' and sync is None:
```

Comment:
Better check for `and not sync` instead of `and sync is None`.

The way it is now fails for calls alike `self._parse_operation(operation, sync=False)`